trin-core/src/portalnet/types/metric.rs

```diff
@@ -1,22 +1,73 @@
+use std::fmt;
+use std::ops::Deref;
+
 use ethereum_types::U256;
 
+/// Represents a distance between two keys in the DHT key space.
+#[derive(Copy, Clone, PartialEq, Eq, Default, PartialOrd, Ord, Debug)]
+pub struct Distance(U256);
+
+impl fmt::Display for Distance {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl Distance {
+    /// The maximum value.
+    pub const MAX: Self = Self(U256::MAX);
+
+    /// Returns the integer base-2 logarithm of `self`.
+    ///
+    /// Returns `None` is `self` is zero, because the logarithm of zero is undefined. Otherwise,
+    /// returns `Some(log2)` where `log2` is in the range [1, 256].
+    pub fn log2(&self) -> Option<u64> {
```

Comment:
I think a `Result` could make even more sense here. Like I would expect an `Err` if trying to divide by zero, more than a `None`.

Also, is `usize` preferable here? If it was a 32-bit system, it seems like that would be fine. I have no real opinion on this part.

Reply:
I would push back in favor of `None`, because it captures the idea of "undefined". If we return a `Result`, that is another error that we have to create, or if we return the unit type `()` as the `Err` variant, then that is equivalent to `None`.

I would also note that the Rust primitive types return `None` for [`checked_div`](https://doc.rust-lang.org/std/primitive.usize.html#method.checked_div) on divide by zero, so it seems `None` is more idiomatic.

Good suggestion on `usize`. `U256::leading_zeros` returns a `u32`, so there are no conversion problems.

Reply:
> I would also note that the Rust primitive types return `None` for [`checked_div`](https://doc.rust-lang.org/std/primitive.usize.html#method.checked_div) on divide by zero, so it seems `None` is more idiomatic.

An excellent reason to keep it as `Option`. 👍🏻